app/src/main/java/org/buffer/android/components/DialogFactory.kt

```diff
@@ -343,26 +342,24 @@ object DialogFactory {
     }
 
     fun showLimitedLengthInputDialog(
-            context: Context, maxLength: Int,
-            @StringRes titleResource: Int,
-            @StringRes messageResource: Int,
-            @StringRes positive: Int,
-            @StringRes negative: Int,
-            inputListener: InputListener? = null,
-            defaultText: String? = null
+        context: Context,
+        maxLength: Int,
+        titleResource: String,
+        messageResource: String,
+        positive: String,
+        negative: String,
+        inputListener: InputListener? = null,
+        defaultText: String? = null
     ): AlertDialog {
         val builder = MaterialAlertDialogBuilder(context)
-                .setTitle(titleResource)
-                .setMessage(messageResource)
+            .setTitle(titleResource)
+            .setMessage(messageResource)
 
         val layout = LayoutInflater.from(context)
                 .inflate(R.layout.view_limited_length_input, null)
         val editText = layout.findViewById<EditText>(R.id.input)
-        val counterView = layout.findViewById<CounterView>(R.id.view_counter)
+        val inputLayout = layout.findViewById<TextInputLayout>(R.id.textContainer)
         if (defaultText != null) editText.setText(defaultText)
-        counterView.charactersRemainingUntilCounterDisplay = (maxLength * 0.15).toInt()
```

Comment:
i've removed the counter view and utilised the textinputlayout error state for a more native feel